ghcide/src/Development/IDE/Core/ProgressReporting.hs

```diff
@@ -42,38 +48,60 @@ data ProgressEvent
   | ProgressCompleted
   | ProgressStarted
 
-data ProgressReporting m = ProgressReporting
-  { progressUpdate :: ProgressEvent -> m (),
-    inProgress     :: forall a. NormalizedFilePath -> m a -> m a,
-    -- ^ see Note [ProgressReporting API and InProgressState]
-    progressStop   :: IO ()
+data ProgressReportingNoTrace m = ProgressReportingNoTrace
+  { progressUpdateI :: ProgressEvent -> m (),
+    progressStopI   :: IO ()
     -- ^ we are using IO here because creating and stopping the `ProgressReporting`
     -- is different from how we use it.
   }
 
+data ProgressReporting m = ProgressReporting
```

Comment:
I would name the other one `ProgressReporting` and call this `PerFileProgressReporting` or something?

Reply:
make sense, I like this naming